### PR TITLE
[SPARK-12212][ML][DOC] Clarifies the difference between spark.ml, spark.mllib and mllib in the documentation.

### PR DESCRIPTION
**Pull request by thunterdb**
Replaces a number of occurences of `MLlib` in the documentation that were meant to refer to the `spark.mllib` package instead. It should clarify for new users the difference between `spark.mllib` (the package) and MLlib (the umbrella project for ML in spark).

It also removes some files that I forgot to delete with #10207

### CODE REVIEW

**Comment by jkbradley** [comment]
Early comment: Maybe we should reinstate all existing .md files but replace the contents with redirection links to the relevant new parts of the guide.  Websites might have links which will be broken by these changes.

**Comment by jkbradley** [comment]
Related to my earlier comment, it might be good to rename the ml-intro file to ml-guide; there isn't a need to rename it.

File: docs/ml-classification-regression.md [hunk 1]
```diff
@@ -27,10 +27,10 @@ displayTitle: Classification and regression in spark.ml
 * This will become a table of contents (this text will be scraped).
 {:toc}
 
-In MLlib, we implement popular linear methods such as logistic
+In `spark.ml`, we implement popular linear methods such as logistic
 regression and linear least squares with $L_1$ or $L_2$ regularization.
 Refer to [the linear methods in mllib](mllib-linear-methods.html) for
-details.  In `spark.ml`, we also include Pipelines API for [Elastic
+details about implementation and tuning.  We also include a Dataframe API for [Elastic
```
Comment (jkbradley) [inline docs/ml-classification-regression.md]: "DataFrame"  (F capitalized) (elsewhere too)

File: docs/mllib-feature-extraction.md [hunk 2]
```diff
@@ -1,7 +1,7 @@
 ---
 layout: global
-title: Feature Extraction and Transformation - MLlib
-displayTitle: <a href="mllib-guide.html">MLlib</a> - Feature Extraction and Transformation
+title: Feature Extraction and Transformation
```
Comment (jkbradley) [inline docs/mllib-feature-extraction.md]: Add ``` - spark.mllib``` to end of title?

File: docs/mllib-evaluation-metrics.md [hunk 3]
```diff
@@ -1,20 +1,20 @@
 ---
 layout: global
-title: Evaluation Metrics - MLlib
-displayTitle: <a href="mllib-guide.html">MLlib</a> - Evaluation Metrics
+title: Evaluation Metrics
```
Comment (jkbradley) [inline docs/mllib-evaluation-metrics.md]: ```spark.mllib``` in title

**Comment by jkbradley** [comment]
Done with review.
One final comment: I'd prefer adding spark.ml or spark.mllib to the end of both the title and displayTitle.  I like having it in displayTitle since that provides the clearest signal about which page the user is viewing.

**Comment by thunterdb** [comment]
@jkbradley I agree with you. Note that when you use the doc, the current section appears in bold in the side menu (and the submenu gets expandend). This is why I did not include it again, but I do not have a strong opinion on this point:

screen shot 2015-12-09 at 5 14 18 pm

**Comment by jkbradley** [comment]
OK then I'll still vote for including it : )

File: docs/ml-classification-regression.md [hunk 4]
```diff
@@ -27,10 +27,10 @@ displayTitle: Classification and regression in spark.ml
 * This will become a table of contents (this text will be scraped).
 {:toc}
 
-In MLlib, we implement popular linear methods such as logistic
+In `spark.ml`, we implement popular linear methods such as logistic
```
Comment (srowen) [inline docs/ml-classification-regression.md]: spark.mllib right?

Comment (jkbradley) [inline docs/ml-classification-regression.md]: Not this file; the "ml-" prefix ones are for spark.ml.  (It's true the functionality is almost the same currently, but it's a bit different and will diverge more.)

File: docs/ml-classification-regression.md (hunk 1 above)
Comment (thunterdb) [inline docs/ml-classification-regression.md]: Corrected everywhere in the documentation

File: docs/mllib-feature-extraction.md (hunk 2 above)
Comment (thunterdb) [inline docs/mllib-feature-extraction.md]: Done everywhere

File: docs/ml-classification-regression.md (hunk 4 above)
Comment (srowen) [inline docs/ml-classification-regression.md]: I see the purpose now. It was the old MLlib text, but a lot of it still applies. The distinction is removed.

**Comment by thunterdb** [comment]
@jkbradley done with changes, let me know what you think.

**Comment by jkbradley** [comment]
In addition to moving ml-intro back to ml-guide, it'd be nice if the sidebar had links back to the main spark.ml and spark.mllib pages.  That could be done in a separate JIRA/PR, if you prefer.

**Comment by thunterdb** [comment]
@jkbradley done:
screen shot 2015-12-10 at 11 19 30 am

File: docs/ml-survival-regression.md [hunk 5]
```diff
@@ -1,7 +1,7 @@
 ---
 layout: global
-title: Survival Regression - ML
-displayTitle: <a href="ml-guide.html">ML</a> - Survival Regression
+title: Survival Regression - spark.ml
+displayTitle: Survival Regression - spark.ml
```
Comment (jkbradley) [inline docs/ml-survival-regression.md]: This doc should now be a redirect to the ml-classification-regression.html#survival-regression section.

Also, it looks like some of the math renders incorrectly, but let's fix that in a follow-up.

**Comment by jkbradley** [comment]
That's the only remaining issue I found.  I checked against the Spark 1.5 doc links as well.

**Comment by jkbradley** [comment]
LGTM pending tests.  Thanks!  (not sure what the deal is with survival regression math; I probably am missing some library on my computer.)

**Comment by jkbradley** [comment]
Merging with master and branch-1.6
Thank you!

**Pull request closed without merging**